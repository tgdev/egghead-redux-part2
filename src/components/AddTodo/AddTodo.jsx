import React from 'react';
import PropTypes from 'prop-types';

import { addTodo } from 'actions/todo.actions';

const AddTodo = ({ dispatch }) => {
  let input;

  return [
    <input key='add-todo-input' type="text" ref={node => { input = node; }} />,
    <button
      key='add-todo-btn'
      onClick={() => {
        dispatch(addTodo(input.value));
        input.value = '';
      }}>
      Add Todo
    </button>
  ];
};

AddTodo.propTypes = {
  dispatch: PropTypes.func
};

export default AddTodo;
