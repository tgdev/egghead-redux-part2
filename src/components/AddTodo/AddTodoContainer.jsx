import { connect } from 'react-redux';
import AddTodo from './AddTodo';

const AddTodoContainer = connect()(AddTodo);

export default AddTodoContainer;
