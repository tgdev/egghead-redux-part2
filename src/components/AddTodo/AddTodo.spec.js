import React from 'react';
import { shallow } from 'enzyme';

import AddTodo from './AddTodo';

describe('AddTodo component', () => {
  it('renders AddTodo correctly', () => {
    const wrapper = shallow(<AddTodo />);
    expect(wrapper).toMatchSnapshot();
  });

});
