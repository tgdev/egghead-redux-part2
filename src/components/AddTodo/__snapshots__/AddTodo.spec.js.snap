// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo component renders AddTodo correctly 1`] = `
Array [
  <input
    key="add-todo-input"
    type="text"
  />,
  <button
    key="add-todo-btn"
    onClick={[Function]}
  />,
]
`;
